backend/src/main/java/com/festago/fcm/application/FcmClient.java

```diff
@@ -0,0 +1,10 @@
+package com.festago.fcm.application;
+
+import com.festago.fcm.domain.FCMChannel;
+import com.festago.fcm.dto.FcmPayload;
+import java.util.List;
+
+public interface FcmClient {
+    
+    void sendAll(List<String> tokens, FCMChannel channel, FcmPayload fcmPayload);
```

Comment:
```suggestion
    @Async
    void sendAll(List<String> tokens, FCMChannel channel, FcmPayload fcmPayload);
```
이렇게 달면 안돼나요? [overflow](https://stackoverflow.com/questions/9109126/spring-async-annotation-on-interface-methods)

Reply:
저친구 놀랍게도 sendAll은 Async 메서드가 아닙니다..!
그 이유는 QR 상태 변경용 FCM에서는 이미 비동기 메서드에서 해당 메서드를 호출하기때문에
해당 메서드에 `@Async` 달면 두 번 Async 처리가 되더라구요!
그래도 큰 문제는 없지만, 굳이 두 번 Async를 탈 필요는 없어서 저렇게 진행했습니다

Reply:
그러면
```
@Async
    public void sendEntryAlert(Long id) {
        EntryAlert entryAlert = entryAlertRepository.findByIdAndStatusForUpdate(id, AlertStatus.PENDING)
            .orElseThrow(() -> new ConflictException(ErrorCode.ALREADY_ALERT));
        List<String> tokens = findFcmTokens(entryAlert);
        log.info("EntryAlert 전송 시작 / entryAlertId: {} / to {} devices", id, tokens.size());
        taskExecutor.execute(() -> fcmClient.sendAll(tokens, FCMChannel.ENTRY_ALERT, FcmPayload.entryAlert()));
        entryAlert.request();
    }
```
마찬가지로 요기도
```
@Async
    public void sendEntryAlert(Long id) {
        EntryAlert entryAlert = entryAlertRepository.findByIdAndStatusForUpdate(id, AlertStatus.PENDING)
            .orElseThrow(() -> new ConflictException(ErrorCode.ALREADY_ALERT));
        List<String> tokens = findFcmTokens(entryAlert);
        log.info("EntryAlert 전송 시작 / entryAlertId: {} / to {} devices", id, tokens.size());
        fcmClient.sendAll(tokens, FCMChannel.ENTRY_ALERT, FcmPayload.entryAlert());
        entryAlert.request();
    }
```
이렇게 하면 안되는 것인가용?

Reply:
그래도 되긴 하는데... `entryAlertRepository.findByIdAndStatusForUpdate()` 메서드가 락을 걸기도 하고, 트랜잭션 범위안에 있어서 이건 비동기로 재호출을 한 번 더 해야될 것 같네요

Reply:
아아 x 락 범위가 있었네요 굿굿